### ISSUE TITLE
Uploading wheel fail due to `The description failed to render in the default format of reStructuredText.`

### ISSUE THREAD

**Issue by 4383** [pypa/warehouse#8791]
**Describe the bug**

Since few days we faced uploading failures during upload of https://opendev.org/openstack/ansible-role-redhat-subscription/ .

The README of this project is at the markdown format. 
One year ago we already fixed an issue related to the upload of the markdown format by providing [`long_description_content_type='text/markdown'`](https://opendev.org/openstack/ansible-role-redhat-subscription/commit/fceb51c66eb343a7cd20891da8728c65ff703ce7) in our `setup.py` and it fixed our upload.

Since this fix was merged we released successfully two new versions:
- https://pypi.org/project/ansible-role-redhat-subscription/1.0.4/
- https://pypi.org/project/ansible-role-redhat-subscription/1.1.0/
 
But since few days we encounter the following issue:

```
The description failed to render in the default format of reStructuredText. See https://pypi.org/help/#description-content-type for more information.
```

From our perspective we didn't see significant changes who could explain why this issue re-appear suddenly.

1.1.0 uploaded successfully on 2020-05-21, that probably relied on warehouse using readme-renderer 26.0 from 2020-04-23. Since two new versions of readme-renderer have been released and maybe warehouse rely on one of them. Something changed between 1.1.0 and 1.1.1 (our version that fail).

We diffed the wheel metadata files between 1.1.0 and 1.1.1 and the only differences were the version and a few rows added/changed in that table in the description (https://github.com/openstack/ansible-role-redhat-subscription/blame/master/README.md#L17,L57).

Locally `python setup.py sdist` work as expected and `PKG-INFO` seems ok.

We also tested `readme-renderer` locally and we just 

readme-renderer's 27.0 changelog says: "Add support for align attribute rendering Markdown headers". Maybe a regression crept in there? 

I hope that these info are useful for you.

**My Platform**

- Pygments-2.7.2
- SecretStorage-3.1.2
- bleach-3.2.1
- colorama-0.4.4
- jeepney-0.4.3
- keyring-21.4.0
- packaging-20.4
- pkginfo-1.6.1
- pyparsing-2.4.7
- readme-renderer-28.0
- requests-toolbelt-0.9.1
- rfc3986-1.4.0
- tqdm-4.51.0
- twine-3.2.0
- webencodings-0.5.1

**Additional context**

Logs of our job used to upload our wheel:

```
2020-11-06 13:20:20.458674 | TASK [upload-pypi : Upload wheels and sdist tarballs with twine]
2020-11-06 13:20:24.056351 | localhost | Uploading distributions to https://upload.pypi.org/legacy/
2020-11-06 13:20:24.056874 | localhost | Uploading ansible_role_redhat_subscription-1.1.1-py2.py3-none-any.whl
2020-11-06 13:20:24.056986 | localhost |
2020-11-06 13:20:24.058532 | localhost |   0%|          | 0.00/29.0k [00:00<?, ?B/s]
2020-11-06 13:20:24.058636 | localhost |  28%|██▊       | 8.00k/29.0k [00:00<00:00, 51.3kB/s]
2020-11-06 13:20:24.058737 | localhost | 100%|██████████| 29.0k/29.0k [00:00<00:00, 37.4kB/s]
2020-11-06 13:20:24.058835 | localhost | NOTE: Try --verbose to see response content.
2020-11-06 13:20:24.058930 | localhost | HTTPError: 400 Bad Request from https://upload.pypi.org/legacy/
2020-11-06 13:20:24.059024 | localhost | The description failed to render in the default format of reStructuredText. See https://pypi.org/help/#description-content-type for more information.
2020-11-06 13:20:24.059182 | localhost | ERROR
2020-11-06 13:20:24.059440 | localhost | {
2020-11-06 13:20:24.059542 | localhost |   "delta": "0:00:02.878461",
2020-11-06 13:20:24.059632 | localhost |   "end": "2020-11-06 13:20:23.988985",
2020-11-06 13:20:24.059998 | localhost |   "msg": "non-zero return code",
2020-11-06 13:20:24.060092 | localhost |   "rc": 1,
2020-11-06 13:20:24.060180 | localhost |   "start": "2020-11-06 13:20:21.110524"
2020-11-06 13:20:24.060270 | localhost | }
```

Full logs are available at:
https://zuul.opendev.org/t/openstack/stream/6c2e633da6b3465fa316f437f7b84ef2?logfile=console.log

Thanks for your help

**Comment by di** [comment]
Can you share the actual distribution file (`ansible_role_redhat_subscription-1.1.1-py2.py3-none-any.whl`) you are attempting to upload here?

Can you share the versions of twine, setuptools and wheel used as well?

**Comment by 4383** [comment]
Generated locally:
[ansible_role_redhat_subscription-1.1.1-py2.py3-none-any.zip](https://github.com/pypa/warehouse/files/5502189/ansible_role_redhat_subscription-1.1.1-py2.py3-none-any.zip)

Locally I pull:
- setuptools 40.6.2
- wheel 0.34.2
- twine 3.1.1 

But I need to confirm which versions are in use in our job, I only know that this env use twine 3.2.0.

**Comment by fungi** [comment]
Apologies, on closer examination I think this can be closed:

Between the working and failing uploads, we switched our CI system to install the wheel module from distro packaging rather than from PyPI by default. Since the recent build was performed on an Ubuntu 18.04 LTS instance, it used the available python3-wheel 0.30.0-0.2 available there. As wheel did not introduce support for Python packaging metadata version 2.1 until its 0.31.0 release, the failing upload declared "Metadata-Version: 2.0" in the METADATA file. This seems to cause the "Description-Content-Type: text/markdown" line (first introduced in the metadata 2.1 specification) to be ignored and fall back to assuming the long description is in the reStructuredText default.

Per the above, it looks to me like everything is working as designed. We'll figure out the most effective way to get our release jobs using a more up to date version of wheel. Thanks for the quick response, and sorry for the false alarm!

**Comment by 4383** [comment]
@fungi thanks for the heads up.

@di sorry for the disturb
